### PR TITLE
Upgrade to Vert.x 4.4.9 with backports

### PR DESCRIPTION
**Pull request by jponge**
This is a backport of #39764 on top of Vert.x 4.4.9

### CODE REVIEW

**Comment by jponge** [comment]
I'm having issues locally with native compilation, but this could be due to GraalVM baselines, to be investigated.

**Comment by jponge** [comment]
👋  @zakkak @geoand I'm struggling with this backport, it looks like some Netty classes get build-time initialized while they shouldn't. It's really the Netty version bump that brings these issues on 3.2:

```
Error: Classes that should be initialized at run time got initialized during image building:
 io.netty.buffer.AbstractReferenceCountedByteBuf the class was requested to be initialized at run time (from 'META-INF/native-image/io.netty/netty-buffer/native-image.properties' in 'file:///Volumes/Code/Quarkus/quarkus/integration-tests/resteasy-jackson/target/quarkus-integration-test-resteasy-jackson-3.2.999-SNAPSHOT-native-image-source-jar/lib/io.netty.netty-buffer-4.1.108.Final.jar' with 'io.netty.buffer.AbstractReferenceCountedByteBuf'). To see why io.netty.buffer.AbstractReferenceCountedByteBuf got initialized use --trace-class-initialization=io.netty.buffer.AbstractReferenceCountedByteBuf
io.netty.buffer.ByteBufUtil the class was requested to be initialized at run time (from 'META-INF/native-image/io.netty/netty-buffer/native-image.properties' in 'file:///Volumes/Code/Quarkus/quarkus/integration-tests/resteasy-jackson/target/quarkus-integration-test-resteasy-jackson-3.2.999-SNAPSHOT-native-image-source-jar/lib/io.netty.netty-buffer-4.1.108.Final.jar' with 'io.netty.buffer.ByteBufUtil' and from feature io.quarkus.runner.Feature.beforeAnalysis with 'ByteBufUtil.class'). To see why io.netty.buffer.ByteBufUtil got initialized use --trace-class-initialization=io.netty.buffer.ByteBufUtil
io.netty.buffer.UnpooledUnsafeHeapByteBuf the class was requested to be initialized at run time (subtype of io.netty.buffer.AbstractReferenceCountedByteBuf). To see why io.netty.buffer.UnpooledUnsafeHeapByteBuf got initialized use --trace-class-initialization=io.netty.buffer.UnpooledUnsafeHeapByteBuf
io.netty.buffer.PooledByteBufAllocator the class was requested to be initialized at run time (from 'META-INF/native-image/io.netty/netty-buffer/native-image.properties' in 'file:///Volumes/Code/Quarkus/quarkus/integration-tests/resteasy-jackson/target/quarkus-integration-test-resteasy-jackson-3.2.999-SNAPSHOT-native-image-source-jar/lib/io.netty.netty-buffer-4.1.108.Final.jar' with 'io.netty.buffer.PooledByteBufAllocator' and from feature io.quarkus.runner.Feature.beforeAnalysis with 'PooledByteBufAllocator.class'). To see why io.netty.buffer.PooledByteBufAllocator got initialized use --trace-class-initialization=io.netty.buffer.PooledByteBufAllocator
io.netty.buffer.UnpooledHeapByteBuf the class was requested to be initialized at run time (subtype of io.netty.buffer.AbstractReferenceCountedByteBuf). To see why io.netty.buffer.UnpooledHeapByteBuf got initialized use --trace-class-initialization=io.netty.buffer.UnpooledHeapByteBuf
io.netty.buffer.UnpooledDirectByteBuf the class was requested to be initialized at run time (subtype of io.netty.buffer.AbstractReferenceCountedByteBuf). To see why io.netty.buffer.UnpooledDirectByteBuf got initialized use --trace-class-initialization=io.netty.buffer.UnpooledDirectByteBuf
io.netty.buffer.UnpooledByteBufAllocator$InstrumentedUnpooledUnsafeDirectByteBuf the class was requested to be initialized at run time (subtype of io.netty.buffer.AbstractReferenceCountedByteBuf). To see why io.netty.buffer.UnpooledByteBufAllocator$InstrumentedUnpooledUnsafeDirectByteBuf got initialized use --trace-class-initialization=io.netty.buffer.UnpooledByteBufAllocator$InstrumentedUnpooledUnsafeDirectByteBuf
io.netty.buffer.UnpooledByteBufAllocator$InstrumentedUnpooledUnsafeHeapByteBuf the class was requested to be initialized at run time (subtype of io.netty.buffer.AbstractReferenceCountedByteBuf). To see why io.netty.buffer.UnpooledByteBufAllocator$InstrumentedUnpooledUnsafeHeapByteBuf got initialized use --trace-class-initialization=io.netty.buffer.UnpooledByteBufAllocator$InstrumentedUnpooledUnsafeHeapByteBuf
io.netty.buffer.UnpooledUnsafeDirectByteBuf the class was requested to be initialized at run time (subtype of io.netty.buffer.AbstractReferenceCountedByteBuf). To see why io.netty.buffer.UnpooledUnsafeDirectByteBuf got initialized use --trace-class-initialization=io.netty.buffer.UnpooledUnsafeDirectByteBuf
To see how the classes got initialized, use --trace-class-initialization=io.netty.buffer.AbstractReferenceCountedByteBuf,io.netty.buffer.ByteBufUtil,io.netty.buffer.UnpooledUnsafeHeapByteBuf,io.netty.buffer.PooledByteBufAllocator,io.netty.buffer.UnpooledHeapByteBuf,io.netty.buffer.UnpooledDirectByteBuf,io.netty.buffer.UnpooledByteBufAllocator$InstrumentedUnpooledUnsafeDirectByteBuf,io.netty.buffer.UnpooledByteBufAllocator$InstrumentedUnpooledUnsafeHeapByteBuf,io.netty.buffer.UnpooledUnsafeDirectByteBuf
com.oracle.svm.core.util.UserError$UserException: Classes that should be initialized at run time got initialized during image building:
 io.netty.buffer.AbstractReferenceCountedByteBuf the class was requested to be initialized at run time (from 'META-INF/native-image/io.netty/netty-buffer/native-image.properties' in 'file:///Volumes/Code/Quarkus/quarkus/integration-tests/resteasy-jackson/target/quarkus-integration-test-resteasy-jackson-3.2.999-SNAPSHOT-native-image-source-jar/lib/io.netty.netty-buffer-4.1.108.Final.jar' with 'io.netty.buffer.AbstractReferenceCountedByteBuf'). To see why io.netty.buffer.AbstractReferenceCountedByteBuf got initialized use --trace-class-initialization=io.netty.buffer.AbstractReferenceCountedByteBuf
io.netty.buffer.ByteBufUtil the class was requested to be initialized at run time (from 'META-INF/native-image/io.netty/netty-buffer/native-image.properties' in 'file:///Volumes/Code/Quarkus/quarkus/integration-tests/resteasy-jackson/target/quarkus-integration-test-resteasy-jackson-3.2.999-SNAPSHOT-native-image-source-jar/lib/io.netty.netty-buffer-4.1.108.Final.jar' with 'io.netty.buffer.ByteBufUtil' and from feature io.quarkus.runner.Feature.beforeAnalysis with 'ByteBufUtil.class'). To see why io.netty.buffer.ByteBufUtil got initialized use --trace-class-initialization=io.netty.buffer.ByteBufUtil
io.netty.buffer.UnpooledUnsafeHeapByteBuf the class was requested to be initialized at run time (subtype of io.netty.buffer.AbstractReferenceCountedByteBuf). To see why io.netty.buffer.UnpooledUnsafeHeapByteBuf got initialized use --trace-class-initialization=io.netty.buffer.UnpooledUnsafeHeapByteBuf
io.netty.buffer.PooledByteBufAllocator the class was requested to be initialized at run time (from 'META-INF/native-image/io.netty/netty-buffer/native-image.properties' in 'file:///Volumes/Code/Quarkus/quarkus/integration-tests/resteasy-jackson/target/quarkus-integration-test-resteasy-jackson-3.2.999-SNAPSHOT-native-image-source-jar/lib/io.netty.netty-buffer-4.1.108.Final.jar' with 'io.netty.buffer.PooledByteBufAllocator' and from feature io.quarkus.runner.Feature.beforeAnalysis with 'PooledByteBufAllocator.class'). To see why io.netty.buffer.PooledByteBufAllocator got initialized use --trace-class-initialization=io.netty.buffer.PooledByteBufAllocator
io.netty.buffer.UnpooledHeapByteBuf the class was requested to be initialized at run time (subtype of io.netty.buffer.AbstractReferenceCountedByteBuf). To see why io.netty.buffer.UnpooledHeapByteBuf got initialized use --trace-class-initialization=io.netty.buffer.UnpooledHeapByteBuf
io.netty.buffer.UnpooledDirectByteBuf the class was requested to be initialized at run time (subtype of io.netty.buffer.AbstractReferenceCountedByteBuf). To see why io.netty.buffer.UnpooledDirectByteBuf got initialized use --trace-class-initialization=io.netty.buffer.UnpooledDirectByteBuf
io.netty.buffer.UnpooledByteBufAllocator$InstrumentedUnpooledUnsafeDirectByteBuf the class was requested to be initialized at run time (subtype of io.netty.buffer.AbstractReferenceCountedByteBuf). To see why io.netty.buffer.UnpooledByteBufAllocator$InstrumentedUnpooledUnsafeDirectByteBuf got initialized use --trace-class-initialization=io.netty.buffer.UnpooledByteBufAllocator$InstrumentedUnpooledUnsafeDirectByteBuf
io.netty.buffer.UnpooledByteBufAllocator$InstrumentedUnpooledUnsafeHeapByteBuf the class was requested to be initialized at run time (subtype of io.netty.buffer.AbstractReferenceCountedByteBuf). To see why io.netty.buffer.UnpooledByteBufAllocator$InstrumentedUnpooledUnsafeHeapByteBuf got initialized use --trace-class-initialization=io.netty.buffer.UnpooledByteBufAllocator$InstrumentedUnpooledUnsafeHeapByteBuf
io.netty.buffer.UnpooledUnsafeDirectByteBuf the class was requested to be initialized at run time (subtype of io.netty.buffer.AbstractReferenceCountedByteBuf). To see why io.netty.buffer.UnpooledUnsafeDirectByteBuf got initialized use --trace-class-initialization=io.netty.buffer.UnpooledUnsafeDirectByteBuf
To see how the classes got initialized, use --trace-class-initialization=io.netty.buffer.AbstractReferenceCountedByteBuf,io.netty.buffer.ByteBufUtil,io.netty.buffer.UnpooledUnsafeHeapByteBuf,io.netty.buffer.PooledByteBufAllocator,io.netty.buffer.UnpooledHeapByteBuf,io.netty.buffer.UnpooledDirectByteBuf,io.netty.buffer.UnpooledByteBufAllocator$InstrumentedUnpooledUnsafeDirectByteBuf,io.netty.buffer.UnpooledByteBufAllocator$InstrumentedUnpooledUnsafeHeapByteBuf,io.netty.buffer.UnpooledUnsafeDirectByteBuf
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.UserError.abort(UserError.java:73)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ProvenSafeClassInitializationSupport.checkDelayedInitialization(ProvenSafeClassInitializationSupport.java:277)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.duringAnalysis(ClassInitializationFeature.java:219)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$10(NativeImageGenerator.java:786)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:90)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$11(NativeImageGenerator.java:786)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.AbstractAnalysisEngine.runAnalysis(AbstractAnalysisEngine.java:181)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:783)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:592)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:550)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:539)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:721)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.start(NativeImageGeneratorRunner.java:143)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:98)
```

**Comment by geoand** [comment]
I can take a look on Monday

**Comment by jponge** [comment]
It's picking `native-image.properties` metadata in `netty-buffer`:

```
io.netty.buffer.PooledByteBufAllocator the class was requested to be initialized at run time (from 'META-INF/native-image/io.netty/netty-buffer/native-image.properties' in 'file:///Volumes/Code/Quarkus/quarkus/integration-tests/resteasy-jackson/target/quarkus-integration-test-resteasy-jackson-3.2.999-SNAPSHOT-native-image-source-jar/lib/io.netty.netty-buffer-4.1.108.Final.jar' with 'io.netty.buffer.PooledByteBufAllocator' and from feature io.quarkus.runner.Feature.beforeAnalysis with 'PooledByteBufAllocator.class'). io.netty.handler.ssl.JdkSslServerContext caused initialization of this class with the following trace:
	at io.netty.buffer.PooledByteBufAllocator.<clinit>(PooledByteBufAllocator.java:38)
	at io.netty.buffer.ByteBufUtil.<clinit>(ByteBufUtil.java:88)
	at io.netty.buffer.Unpooled.copiedBufferAscii(Unpooled.java:615)
	at io.netty.buffer.Unpooled.copiedBuffer(Unpooled.java:586)
	at io.netty.handler.ssl.PemReader.readCertificates(PemReader.java:98)
	at io.netty.handler.ssl.SslContext.toX509Certificates(SslContext.java:1260)
	at io.netty.handler.ssl.JdkSslServerContext.checkIfWrappingTrustManagerIsSupported(JdkSslServerContext.java:80)
	at io.netty.handler.ssl.JdkSslServerContext.<clinit>(JdkSslServerContext.java:65)
```

**Comment by geoand** [comment]
We can exclude this file, there is a build item (I think the Oracle jdbc extension uses it)

**Comment by geoand** [comment]
`ExcludeConfigBuildItem` is what I had in mind

**Comment by jponge** [comment]
I'll get back to this tomorrow (PTO) - I had tried excluding those metadata but ended up going circles / it looks like some SSL-related code is triggering those build-time initialisations.

**Comment by jponge** [comment]
Tracing class initializations always point me to `JdkSslServerContext`, as in:

```
at io.netty.buffer.PooledByteBufAllocator.<clinit>(PooledByteBufAllocator.java:38)
	at io.netty.buffer.ByteBufUtil.<clinit>(ByteBufUtil.java:88)
	at io.netty.buffer.Unpooled.copiedBufferAscii(Unpooled.java:615)
	at io.netty.buffer.Unpooled.copiedBuffer(Unpooled.java:586)
	at io.netty.handler.ssl.PemReader.readCertificates(PemReader.java:98)
	at io.netty.handler.ssl.SslContext.toX509Certificates(SslContext.java:1260)
	at io.netty.handler.ssl.JdkSslServerContext.checkIfWrappingTrustManagerIsSupported(JdkSslServerContext.java:80)
	at io.netty.handler.ssl.JdkSslServerContext.<clinit>(JdkSslServerContext.java:65)
```

I haven't seen any obvious change in the Netty / Vert.x extensions between `3.2` and `main`.

**Comment by jponge** [comment]
@geoand that's what I ended up doing but we still have the gRPC and cache native tests failing, but not on native compilation. I don't know if it's related.

**Comment by geoand** [comment]
Indeed it's hard to tell...

**Comment by jponge** [comment]
I've been able to test `integration-tests/infinispan-client ` successfully locally.

It looks like there is a more serious issue with `integration-tests/grpc-plain-text-gzip` as the produced binaries fail to respond correctly:

```
[ERROR] Failures:
[ERROR]   HelloWorldEndpointIT>HelloWorldEndpointTestBase.testHelloWorldServiceUsingBlockingStub:22
Expecting actual:
  "{"details":"Error id 887631a4-cfdf-4583-852e-8fa9a180fda3-1","stack":""}"
to start with:
  "Hello neo"

[ERROR]   HelloWorldEndpointIT>HelloWorldEndpointTestBase.testHelloWorldServiceUsingMutinyStub:29
Expecting actual:
  ""
to start with:
  "Hello neo-mutiny"

[ERROR] Errors:
[ERROR]   HelloWorldServiceIT>HelloWorldServiceTestBase.testHelloWorldServiceUsingBlockingStub:47 » StatusRuntime CANCELLED: Failed to read message.
[ERROR]   HelloWorldServiceIT.testHelloWorldServiceUsingMutinyStub » StatusRuntime CANCELLED: Failed to read message.
[ERROR]   VertxHelloWorldServiceIT>HelloWorldServiceTestBase.testHelloWorldServiceUsingBlockingStub:47 » StatusRuntime INTERNAL: No value received for unary call
[ERROR]   VertxHelloWorldServiceIT>HelloWorldServiceTestBase.testHelloWorldServiceUsingMutinyStub:55 » Timeout
[INFO]
[ERROR] Tests run: 6, Failures: 2, Errors: 4, Skipped: 0
```

I have spotted the characteristic warning where a heuristic fails to handle an `Unsafe` field:

```
Warning: RecomputeFieldValue.FieldOffset automatic substitution failed. The automatic substitution registration was attempted because a call to sun.misc.Unsafe.objectFieldOffset(Field) was detected in the static initializer of com.google.protobuf.UnsafeUtil. Detailed failure reason(s): The argument of sun.misc.Unsafe.objectFieldOffset(java.lang.reflect.Field) is not a constant value or a field load that can be constant-folded., Could not determine the field where the value produced by the call to sun.misc.Unsafe.objectFieldOffset(Field) for the field offset computation is stored. The call is not directly followed by a field store or by a sign extend node followed directly by a field store.
```

/cc @zakkak it's not related to your efforts wrt `UnsafeAccessedFieldBuildItem` (it's a 3.2 backport branch here) but that might be of interest to you as well.

**Comment by jponge** [comment]
Hi @alesj, I'd need some help on backporting gRPC changes to 3.2 with a Netty/Vert.x bump.

I've backported a substitution so `com.google.protobuf.UnsafeUtil` doesn't annoy us, but the IT tests in `integration-tests/grpc-plain-text-gzip` still fail.

I haven't bumped the `grpc.version` and protobuf-related versions in the POM, if they match those from `main` then I get compilation errors because of internal API breaking changes 

Do you have any idea? 🙏

**Comment by alesj** [comment]
@jponge how do I reproduce this compilation errors?

**Comment by jponge** [comment]
@alesj you can checkout this branch locally, do a `./mvnw -Dquickly`, then run `./mvnw clean verify -f integration-tests/grpc-plain-text-gzip -Pnative` (GraalVM 17 or 22)

**Comment by matzew** [comment]
Thanks

**Comment by jponge** [comment]
@alesj I realise I did not answer your question 🤣 

Here's what you need to do, aligning the gRPC/protobuf to what's in `main`:

```diff
diff --git a/pom.xml b/pom.xml
index e78e18008e2..7a80d6b53a5 100644
--- a/pom.xml
+++ b/pom.xml
@@ -72,11 +72,11 @@
         <kubernetes-client.version>6.7.2</kubernetes-client.version> <!-- Please check with Java Operator SDK team before updating -->

         <!-- Make sure to check compatibility between these 2 gRPC components before upgrade -->
-        <grpc.version>1.56.0</grpc.version> <!-- when updating, verify if com.google.auth should not be updated too -->
+        <grpc.version>1.62.2</grpc.version> <!-- when updating, verify if com.google.auth should not be updated too -->
         <grpc-jprotoc.version>1.2.1</grpc-jprotoc.version>
-        <protoc.version>3.22.0</protoc.version>
+        <protoc.version>3.25.0</protoc.version>
         <protobuf-java.version>${protoc.version}</protobuf-java.version>
-        <proto-google-common-protos.version>2.23.0</proto-google-common-protos.version>
+        <proto-google-common-protos.version>2.36.0</proto-google-common-protos.version>

         <!-- TestNG version: we don't enforce it in the BOM as it is mostly used in the MP TCKs and we need to use the version from the TCKs -->
         <testng.version>7.4.0</testng.version>
```

**Comment by zakkak** [comment]
> I've been able to test `integration-tests/infinispan-client ` successfully locally.

@jponge how did you fix this?

Backporting https://github.com/quarkusio/quarkus/pull/37384 seems to fix both `integration-tests/infinispan-client` and `integration-tests/grpc-plain-text-gzip`

**Comment by jponge** [comment]
@zakkak I didn't fix the Infinispan client test, it just worked on my machine ™️ 

I'm going to test backporting https://github.com/quarkusio/quarkus/pull/37384

**Comment by zakkak** [comment]
> @zakkak how did you fix `integration-tests/grpc-plain-text-gzip`? The change fixes only the Infinispan extension.

It just worked after the patch. CI seems happy about it as well. Note that this is without the grpc update mentioned in https://github.com/quarkusio/quarkus/pull/39788#issuecomment-2034304266

**Comment by jponge** [comment]
@zakkak My current branch is the same as this PR (without any gRPC update) and still get issues in native with `integration-tests/grpc-plain-text-gzip`:

```
[ERROR] Failures:
[ERROR]   HelloWorldEndpointIT>HelloWorldEndpointTestBase.testHelloWorldServiceUsingBlockingStub:22
Expecting actual:
  "{"details":"Error id 1bdd53e3-10d3-4d34-a1ef-05148bebcdb0-1","stack":""}"
to start with:
  "Hello neo"

[ERROR]   HelloWorldEndpointIT>HelloWorldEndpointTestBase.testHelloWorldServiceUsingMutinyStub:29
Expecting actual:
  ""
to start with:
  "Hello neo-mutiny"

[ERROR] Errors:
[ERROR]   HelloWorldServiceIT>HelloWorldServiceTestBase.testHelloWorldServiceUsingBlockingStub:47 » StatusRuntime CANCELLED: Failed to read message.
[ERROR]   HelloWorldServiceIT.testHelloWorldServiceUsingMutinyStub » StatusRuntime CANCELLED: Failed to read message.
[ERROR]   VertxHelloWorldServiceIT>HelloWorldServiceTestBase.testHelloWorldServiceUsingBlockingStub:47 » StatusRuntime INTERNAL: No value received for unary call
[ERROR]   VertxHelloWorldServiceIT>HelloWorldServiceTestBase.testHelloWorldServiceUsingMutinyStub:55 » Timeout
[INFO]
[ERROR] Tests run: 6, Failures: 2, Errors: 4, Skipped: 0
```

I'm on macOS / `GraalVM 21.0.2`, let me try with a 23 build.

**Comment by jponge** [comment]
@zakkak Ok, so I have an _interesting_ development here:

- `sdk use java 21.0.2-graalce` 🟢 
- `sdk use java 21.0.2-graal` 🔴

**Comment by jponge** [comment]
I just relaunched the CI on JVM Tests / JDK 11, the rest is green.

@gsmet do you prefer a single commit or is the current set of 4 commits ok?

**Comment by zakkak** [comment]
> @zakkak Ok, so I have an _interesting_ development here:
> 
>     * `sdk use java 21.0.2-graalce` 🟢
> 
>     * `sdk use java 21.0.2-graal` 🔴

There was a report about this in 3.9.x as well, it might not be 3.2.x specific. See https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Native.20compilation.20issue.20with.203.2E9.2E2 

Please let me know if we need to make it work with Oracle GraalVM for JDK 17 as well.

**Review by zakkak — APPROVED** [review]
LGTM

**Review by geoand — APPROVED** [review]

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/ServerLimitsConfig.java [hunk 1]
```diff
@@ -41,6 +41,27 @@ public class ServerLimitsConfig {
...
+    @ConfigItem(defaultValue = "1K")
+    public MemorySize maxFormBufferedBytes;
+
+    /**
+     * The maximum number of HTTP request parameters permitted for incoming requests.
+     * <p>
+     * If a client sends more than this number of parameters in a request, the connection is closed.
+     */
+    @ConfigItem(defaultValue = "1000")
+    public int maxParameters;
```
Comment (rsvoboda) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/ServerLimitsConfig.java:63]: @jponge  this config parameter doesn't seem to be used

Comment (jponge) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/ServerLimitsConfig.java:63]: You're right, it might be removed

**Pull request merged**